provider/postgis/postgis.go

```diff
@@ -201,38 +207,43 @@ func BuildDBConfig(cs string) (*pgxpool.Config, error) {
 		"application_name":              "tegola",
 	}
 
-	var hasInit bool
+	var hstore hstoreOID
 	dbconfig.AfterConnect = func(ctx context.Context, conn *pgx.Conn) error {
 		// The AfterConnect call runs everytime a new connection is acquired,
 		// including everytime the connection pool expands. The hstore OID
-		// lookup should only be run once
-		if !hasInit {
-			// as extension hstore does not have a fixed OID we have to get it on connect
-			var hstoreOID uint32
+		// is not constant, so we lookup the OID once per provider and store it.
+		// Extensions have to be registered for every new connection.
+
+		if !hstore.hasInit {
```

Comment:
nice! I like this `hasInit` check.

Reply:
pat yourself on the back for that, it was your idea 😄